Gemfile

```diff
@@ -7,6 +7,11 @@ source "https://rubygems.org"
 # development dependencies will be added by default to the :development group.
 gemspec
 
+group :development, :test do
+  # The Jasmine Selenium runner needs `require: false` to work properly - and
+  # this doesn't seem to be an option in the `gemspec` file.
+  gem "jasmine_selenium_runner", require: false
```

Comment:
While I'm not one of those Ruby devs that consider Gemfile a big no for gems, for consistency we can move this to gemspec as `require: false` is default in gemspec so we don't need to worry abut that. I suggest we add this to this to the [gemspec](https://github.com/alphagov/govuk_publishing_components/blob/master/govuk_publishing_components.gemspec#L32) as `s.add_development_dependency "jasmine_selenium_runner", "~> 3.0.0"`

Reply:
Hmm, this is a tricky and strange one then.

I originally added the `jasmine_selenium_runner` in the gemspec file - but then the tests were not running correctly. It defaulted to trying to find Firefox to run the tests in - which is the default setting. 

With the dependency in the gemspec file, and running `bundle exec rake app:jasmine:ci` the error is:

```
Selenium::WebDriver::Error::WebDriverError: Could not find Firefox binary (os=macosx). Make sure Firefox is installed or set the path manually with Selenium::WebDriver::Firefox::Binary.path=
```

Adding the dependency to the Gemfile meant that the tests ran correctly in headless chrome. 

I thought it was something to do with `require: false` - removing that from the Gemfile resulted in the same 'Can't find Firefox' error.

I'll dig a bit more and see why this could be. There isn't a consistency for Gemfile vs gemspec across the organisation, which makes things a bit trickier. Most of the implementations I've found that get headless Chrome to run Jasmine tests use a Gemfile, so I'll try and see what a gemspec set up does.

Reply:
A fix was found for content-publisher (https://github.com/alphagov/content-publisher/pull/2110) which I have copied into this pull request. All seems to be working 🤞

Reply:
Yay, glad you were able to make use of that 👍

Reply:
By the way - even though gems don't automatically pull in gems, this happens in our test suite because of this line: https://github.com/alphagov/govuk_publishing_components/blob/206cde79934a62577f2c368c8c869ac2933f4d14/spec/dummy/config/application.rb#L16 (which is a bit of a smell as a part of the test suite makes this app behave differently)